File: yuiextz/src/org/zkoss/yuiext/layout/West.java
/* West.java

 {{IS_NOTE
 Purpose:
 
 Description:
 
 History:
 Aug 7, 2007 6:22:32 PM , Created by jumperchen
 }}IS_NOTE

 Copyright (C) 2007 Potix Corporation. All Rights Reserved.

 {{IS_RIGHT
 This program is distributed under GPL Version 2.0 in the hope that
 it will be useful, but WITHOUT ANY WARRANTY.
 }}IS_RIGHT
 */
package org.zkforge.yuiext.layout;

/**
 * This component is a west region.
 * 
 * @author jumperchen
 * 
 */
public class West extends SplitLayoutRegion {

	private int[] _cmargins = new int[] { 0, 2, 2, 0 };

	/**
	 * defaults to {top: 0, left: 2, right: 2, bottom: 0}
	 */
	public int[] getCmargins() {
		return _cmargins;
	}

	public void setCmargins(int top, int left, int right, int bottom) {
		boolean f = false;
		if (_cmargins[0] != top) {
			f = true;
			_cmargins[0] = top;
		}
		if (_cmargins[1] != left) {
			f = true;
			_cmargins[1] = left;
		}
		if (_cmargins[2] != right) {
			f = true;
			_cmargins[2] = right;
		}
		if (_cmargins[3] != bottom) {
			f = true;
			_cmargins[3] = bottom;
		}
		if (f)
			smartUpdate("z.cmargins", _cmargins[0] + "," + _cmargins[1] + ","
					+ _cmargins[2] + "," + _cmargins[3]);
	}

	/**
	 * Returns {@link BasicLayoutRegion#POSITION_WEST}
	 */
	public String getPosition() {
		return POSITION_WEST;
	}

}

File: zul/src/main/resources/web/js/zul/layout/West.ts
/* West.ts

	Purpose:

	Description:

	History:
		Wed Jan  7 12:15:01     2009, Created by jumperchen

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/

/**
 * A west region of a border layout.
 * @defaultValue {@link getZclass}: z-west.
 * @defaultValue {@link getCmargins} is "0,0,0,0"
 */
@zk.WrapClass('zul.layout.West')
export class West extends zul.layout.LayoutRegion {
	/** @internal */
	_sumFlexWidth = true; //indicate shall add this flex width for borderlayout. @See _fixMinFlex in widget.js
	/** @internal */
	_maxFlexHeight = true; //indicate shall check if the maximum flex height for borderlayout. @See _fixMinFlex in widget.js
	override sanchor = 'l';
	/** @internal */
	override _cmargins = [0, 0, 0, 0];

	/**
	 * The height can't be specified in this component because its height is
	 * determined by other region components ({@link North} or {@link South}).
	 */
	override setHeight(height: string): this { // readonly
		return this;
	}

	/**
	 * @returns The value {@link zul.layout.Borderlayout.WEST}.
	 */
	override getPosition(): string {
		return zul.layout.Borderlayout.WEST;
	}

	/**
	 * @returns the size of this region. This method is shortcut for {@link getWidth}.
	 */
	getSize(): string | undefined {
		return this.getWidth();
	}

	/**
	 * Sets the size of this region. This method is shortcut for {@link setWidth}.
	 */
	setSize(size: string): this {
		return this.setWidth(size);
	}

	/** @internal */
	override _ambit2(ambit: zul.layout.LayoutRegionAmbit, mars: zk.Dimension, split: { offsetWidth; offsetHeight }): void {
		ambit.w += split.offsetWidth;
		ambit.h = mars.height;
		ambit.ts = ambit.x + ambit.w + (mars.width - mars.left); // total size;
	}

	/** @internal */
	override _reszSp2(ambit: zul.layout.LayoutRegionAmbit, split: { w; h }): Partial<{ left; top; width; height }> {
		ambit.w -= split.w;
		return {
			left: jq.px0(ambit.x + ambit.w),
			top: jq.px0(ambit.y),
			height: jq.px0(ambit.h)
		};
	}
}